imageloading-core/src/main/java/dev/chrisbanes/accompanist/imageloading/AndroidDrawablePainter.kt

```diff
@@ -0,0 +1,153 @@
+
+            override fun scheduleDrawable(d: Drawable, what: Runnable, time: Long) {
+                handler.postAtTime(what, time)
+            }
+
+            override fun unscheduleDrawable(d: Drawable, what: Runnable) {
+                handler.removeCallbacks(what)
+            }
+        }
+    }
+
+    override fun applyAlpha(alpha: Float): Boolean {
+        drawable.alpha = (alpha * 255).roundToInt().coerceIn(0, 255)
+        return true
+    }
+
+    override fun applyColorFilter(colorFilter: ColorFilter?): Boolean {
+        if (colorFilter != null) {
+            drawable.setTint(colorFilter.color.toArgb())
+            drawable.setTintMode(
+                when (colorFilter.blendMode) {
+                    BlendMode.Clear -> PorterDuff.Mode.CLEAR
+                    BlendMode.Src -> PorterDuff.Mode.SRC
+                    BlendMode.Dst -> PorterDuff.Mode.DST
+                    BlendMode.SrcOver -> PorterDuff.Mode.SRC_OVER
+                    BlendMode.DstOver -> PorterDuff.Mode.DST_OVER
+                    BlendMode.SrcIn -> PorterDuff.Mode.SRC_IN
+                    BlendMode.DstIn -> PorterDuff.Mode.DST_IN
+                    BlendMode.SrcOut -> PorterDuff.Mode.SRC_OUT
+                    BlendMode.DstOut -> PorterDuff.Mode.DST_OUT
+                    BlendMode.SrcAtop -> PorterDuff.Mode.SRC_ATOP
+                    BlendMode.DstAtop -> PorterDuff.Mode.DST_ATOP
+                    BlendMode.Xor -> PorterDuff.Mode.XOR
+                    BlendMode.Screen -> PorterDuff.Mode.SCREEN
+                    BlendMode.Overlay -> PorterDuff.Mode.OVERLAY
+                    BlendMode.Darken -> PorterDuff.Mode.DARKEN
+                    BlendMode.Lighten -> PorterDuff.Mode.LIGHTEN
+                    BlendMode.Multiply -> PorterDuff.Mode.MULTIPLY
+                    else -> error("BlendMode not supported by Drawable")
+                }
+            )
+        } else {
+            drawable.setTintList(null)
+            drawable.setTintMode(null)
+        }
+        return true
+    }
+
+    override fun applyLayoutDirection(layoutDirection: LayoutDirection): Boolean {
+        return DrawableCompat.setLayoutDirection(
+            drawable,
+            when (layoutDirection) {
+                LayoutDirection.Ltr -> View.LAYOUT_DIRECTION_LTR
+                LayoutDirection.Rtl -> View.LAYOUT_DIRECTION_RTL
+            }
+        )
+    }
+
+    override val intrinsicSize: Size get() = drawableSize
+
+    override fun DrawScope.onDraw() {
+        if (!startedAnimatable && drawable is Animatable && !drawable.isRunning) {
+            // If the drawable is Animatable, start it on the first draw
+            drawable.start()
+            startedAnimatable = true
+        }
+
+        drawCanvas { canvas, size ->
+            // Reading this ensures that we invalidate when invalidateDrawable() is called
+            invalidateTick
+
+            drawable.setBounds(0, 0, size.width.toInt(), size.height.toInt())
```

Comment:
Have to set bounds on every draw? Looks like this could be done in `init`?

Reply:
Discussed over chat, we only know the size in `onDraw()`